indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java

```diff
@@ -252,7 +252,7 @@ static void addSnapshotJarToClassPath(
     final Path hdfsPath = new Path(intermediateClassPath, jarFile.getName());
 
     // existing is used to prevent uploading file multiple times in same run.
-    if (!existing.contains(hdfsPath)) {
+    if (!existing.contains(hdfsPath) || !fs.exists(hdfsPath)) {
```

Comment:
@himanshug I changed to version aaf40e6 on master, `hadoop index` fail randomly again.

It seems not related to concurrently running tasks, but related to busy level of the hadoop cluster.

When the hadoop cluster is busier, it succeed more times.
When the hadoop cluster is very free (no other app running), it fail more times.

Maybe when cluster is free, the cluster execute `deletion of tmp jar directory` more quickly, and the 2nd immediately started job cannot find the jar file on the hdfs.

When the cluster is busy, the `deletion of tmp jar directory` is not completely executed, the 2nd immediately started job can find the jar when starting up.

I attach the log for success and fail job below.